apps/src/lab2/projects/ProjectManager.ts

```diff
@@ -286,7 +293,16 @@ export default class ProjectManager {
   private onSaveFail(errorMessage: string, error: Error) {
     this.saveInProgress = false;
     this.executeSaveFailListeners(error);
-    Lab2MetricsReporter.logError(errorMessage, error);
+    if (error.message.includes('409')) {
+      // If this is a conflict, we need to reload the page.
+      // We set forceReloading to true so the client can skip
+      // showing the user a dialog before reload.
+      this.forceReloading = true;
+      reload();
```

Comment:
Can we also log that we are doing a reload?  Hopefully it's safe to do, but there is a potential case where the server continually returns `409` for some other reason, and users experience infinite reloads; we should have a way to be aware of this happening.

Reply:
Makes sense. Do you or @sanchitmalhotra126 do have thoughts on whether this should be an info or warning level log?

Reply:
Maybe warning, since it's undesirable but recoverable behavior?

Reply:
done!